### ISSUE TITLE
Issue with test class parsing

### ISSUE THREAD

**Issue by Kenji776** [cesarParra/apexdocs#50]
Hey there. I just updated to the new version and now I'm having issues when generating documentation for test classes. It seems to be choking on the @isTest annotation. 

```
Running command: apexdocs-generate -s input -t temp\markdown\

5/17/2022, 2:55:43 PM: Parsing error Exception: Error parsing Apex Body. Line 2:0 - no viable alternative at input '@isTest\n/** \n* @Name StripeIntegrationWrapperTest\n* \n* @Date 2/28/2022\n* @Group Stripe Integration\n* @See StripeIntegrationWrapper\n* @Description Test class for StripeIntegrationWrapper\n**/'
5/17/2022, 2:55:44 PM: Parsing error Exception: Error parsing Apex Body. Line 2:0 - no viable alternative at input '@isTest\n/** \n* @Name StripeRESTServiceTest\n* \n* @Date 3/2/2022\n* @Group Stripe Integration\n* @See StripeRESTService\n* @Description Test class for StripeRESTService. Ensures all methos in the class work as expected.\n**/'
5/17/2022, 2:55:44 PM: Parsed 1 files
StripeRESTService - Doc comment parsing error. Level: Type
Comment:
 /**
* @Name StripeRESTService
* @Date 3/2/2022
* @Group Stripe Integration
* @See StripeRESTServiceTest
* @Description Apex class that creates and exposes a REST endpoint by the name of 'Stripe'. [ORG_URL]/services/apexrest/Stripe (case sensitive!)
* 
*
```

**Comment by cesarParra** [comment]
Thanks @Kenji776 

The issue seems to be when the annotation is on top of the doc block, rather than right on top of the class/method it belongs to. I'm investigating

**Comment by cesarParra** [comment]
I've added support to have the annotations before and/or after the doc block in the latest release (v2.2.5). Hopefully that resolves the issue.

**Comment by Kenji776** [comment]
@cesarParra Thank you for the quick update. I am sorry to be a pain, but it looks like there is still a problem. It's saying line 3 has a . which it doesn't like. That may be the . in my email address I have in the author line as that is line 3 of the source file.

`Running command: apexdocs-generate -s input -t temp\markdown\
Markdown files generated successfully
5/18/2022, 12:13:54 PM: Parsed 1 files
StripeRESTService - Doc comment parsing error. Level: Type
Comment:
 /**
* @Name StripeRESTService
* @Author Sample Namehere (sample.namehere@mycompany.com)
* @Date 3/2/2022
* @Group Stripe Integration
* @See StripeRESTServiceTest
* @Description Apex class that creates and exposes a REST endpoint by the name of 'Stripe'. [ORG_URL]/services/apexrest/Stripe (case sensitive!)
* EX https://sfsandbox--customdomain.my.salesforce.com/services/apexrest/Stripe/ in the sandbox. It is intended to be accessed via a connected app using JWT flow but
* it is not required.
*
* Current valid endpoints are
* PUT /Stripe/Customer/     - Creates/Updates Accounts and Contacts from Stripe Customer information (StripeCustomer.cls)
* PUT /Stripe/Subscription/ - Creates/Updates Order, OrderItem, and 360_Subscription__c records from Stripe Subscription information (StripeSubscription.cls)
* PUT /Stripe/Price/        - Creates/Updates Product2 and PricebookEntry information from Stripe Price information. (StripePrice.cls)
* PUT /Stripe/Agreement/    - Creates/Updates User_Agreement__c records from a custom class (StripeAgreement.cls). Note, this is not an actual Stripe object, its just named as such to keep with conventions for this API.
*
* @See https://stripe.com/docs/api/customers/object
* @See https://stripe.com/docs/api/subscriptions/object
* @See https://stripe.com/docs/api/price/object
*
* Each payload JSON is additionally expected to provide an param called mstar_id in the top level of the payload to match the data with its salesforce record. Other additional
* parameters not included by Stripe by default may be required. See each class for details.
*
* Expected usage: Stripe will call this API via it's various webhooks. The expected flow is to first create a customer by calling the /Customer endpoint with a JSON payload that
* will be able to serialize into a (list of) StripeCustomer object(s). Products and Prices can be created before or after a customer as they are not directly related. Similarily
* they are created by calling the /Price endpoint with a JSON payload that can serialize into a (list of) StripePrice object(s). Once the customer and price/product information
* exists subscriptions can be created/updated by calling the /Subscription endpoint again with a JSON payload that can be serialized into a (list of) StripeSubscription object(s).
*
* All endpoints support 'upsert' meaning they will create or update data as needed depending if a record is found that matches it's unique identifier which varies for each object type.
**/
Exception: Error parsing Apex doc. Line 3:51 - mismatched input '.' expecting SPACE`

**Comment by cesarParra** [comment]
Thanks @Kenji776 

The issue for this one is that it thinks that the `@` in the email address is the start of a brand new tag, and it is complaining about the tag name containing a dot.

I'll see what can be done about this.

**Comment by cesarParra** [comment]
@Kenji776 

I've somewhat addressed this in the latest release (v2.3.0). I say somewhat because I had to make some compromises because I wanted to still be able to support single line docs, but it was difficult to distinguish between what was an email address vs. what was a doc tag because of the `@` symbol.

What I did was add support for emails to be embedded within a new inline tag: `{@email EMAIL_ADDRESS}, similar to what the `{@link CLASS_NAME}` does for classes.

So now something like this can be done:
```
/**
 * @description My Class
 * @contact {@email someone@example.com}
 */
class MyClass {}
```

Let me know if that sounds good and if you have any suggestions on things to improve for this.

---

On a different note, from your sample docs I see you have some descriptions that get pretty long, so I've also added support for some HTML tags (`<br>`, `<p></p>`, `<ul></ul>`, and `<li></li>` that might help you with the output, so it's not just one long paragraph, in case you are interested in using that. The release notes has more info: https://github.com/cesarParra/apexdocs/releases/tag/v2.3.0

Thanks for the feedback!
